### DISCUSSION TITLE
⚠️ WARNING: Telegram Review

### DISCUSSION THREAD

**Discussion by mgilangjanuar** [mgilangjanuar/teledrive#60] [Announcements]
Hello TeleDrive team,

On behalf of Telegram Messenger, I notify that TeleDrive is not complying with the Telegram API Terms of Service regarding the precepts detailed below:

1.3. As a client developer, you must make sure that all the basic features of the main Telegram apps function correctly and in an expected way both in your app and when users of your app communicate with other Telegram users. It is forbidden to force users of other Telegram clients to download your app in order to view certain messages and content sent using your app.

1.4. It is forbidden to interfere with the basic functionality of Telegram. This includes but is not limited to: making actions on behalf of the user without the user's knowledge and consent, preventing self-destructing content from disappearing, preventing last seen and online statuses from being displayed correctly, tampering with the 'read' statuses of messages (e.g. implementing a 'ghost mode'), preventing typing statuses from being sent/displayed, etc.

Please apply the respective changes before October 25, 2021, and notify us, or we will have to follow the steps explained in point 4.2.

Best regards,
Victor Espinoza
Telegram Messenger

**Comment by Aqshola** [comment]
Woah, im surprised, so we should implement all basic features of telegram? including deleting message last seen and other things?

**Reply by mgilangjanuar** [reply]
I think this is silly, we're only productizing a single part of their functionality. Actually, this project is in the grey area of their terms tho 🤔

**Reply by Aqshola** [reply]
yeah agreed, actually its more upgrade version of saved message, well i hope they understand it 👍

**Comment by mgilangjanuar** [comment]
I replied:

---

Hello Victor,

We'll give you how TeleDrive works.

1.3. TeleDrive is only an enhancement of Saved Messages that already Telegram have. We improve it as a files manager with search capability and folder management. So, the users will have the same experience like they save their files in the Saved Messages in the Telegram application. **We'll add some help text that explains how it works when users want to upload their files.**

1.4. TeleDrive is not interfering with the basic functionality of Telegram. We're only adding some features in the Saved Messages and productize it as a web application. **We'll add some onboarding for users on the first login to let them know how we did it.**

The words in bold will be our call to action to make TeleDrive comply with your terms. Any additions or misunderstandings on us?

Thank you.

Regards,
M Gilang Januar

**Comment by mgilangjanuar** [comment]
The good news is: we got noticed by Telegram 🤣

**Comment by mgilangjanuar** [comment]
5 hours without reply, I send an email again

---

Hi Victor and Telegram team,

I apologize for interrupting your day. But, we need an answer to this email before we continue the development of TeleDrive. If you want to access the source code, please kindly share your GitHub username (the repository is still in private mode).

It's okay if the idea is rejected by Telegram terms. But, please give us a clear answer.

Thank you.

Regards,
M Gilang Januar

**Comment by mgilangjanuar** [comment]
And the answer 🥁

---
Hello M Gilang Januar,

Thanks for your reply.

Telegram's nature is to be a messaging app, with all its features already contained in the official clients.

When extracting only one feature, there's a violation of our API Terms since not all the basic features of the main Telegram apps are in function (point 1.3).

Best regards,
Víctor Espinoza
Telegram Messenger

**Comment by mgilangjanuar** [comment]
My last shoot:

---

Hello Victor Espinoza,

Thank you for your clear answer.

Actually, we're on it to build in-app messaging feature on TeleDrive! Can you please take a look the attachment of it? :)

Thank you.

Regards,
M Gilang Januar

https://user-images.githubusercontent.com/3146378/137503280-22f6d007-4934-42e9-b9fd-dba5098b0bee.mp4
